src/transcoding.ts

```diff
@@ -293,6 +311,25 @@ export function getFieldNameOnBehavior(
   return {requiredFields, optionalFields};
 }
 
+// This function gets all the fields recursively
+export function getAllFieldNames(
```

Comment:
Are we actually using this function anywhere else? If not, let's not `export` it.